client/src/pages/home/HomePage.test.tsx

```diff
@@ -0,0 +1,73 @@
+import { describe, it } from 'vitest';
+import { fireEvent, waitFor } from '@testing-library/react';
+import { HomePage } from '@/pages';
+import { APIManager } from '@/utils';
+import {
+  AllRootProvidersWrapper,
+  baseAuthContextValues,
+  COLLECTED_PLN_ID,
+  COLLECTED_TOTAL_ID,
+  mockEndpoint,
+  renderWithWrapper,
+} from '@tests/utils';
+
+describe('Testing HomePage', () => {
+  beforeEach(() => {
+    mockEndpoint(`${APIManager.baseAPIRUrl}/stats`, {
+      httpVerb: 'get',
+      status: 200,
+      body: {
+        amount_total_in_PLN: 200,
+        amount_PLN: 100,
+      },
+    });
+  });
+  it.concurrent('Testing proper rendering when superadmin logged in', ({ expect }) => {
+    const { getByTestId, getByText } = renderWithWrapper(
+      <HomePage />,
+      AllRootProvidersWrapper(),
+    );
+    expect(getByText('Wydaj puszkę wolontariuszowi')).toBeInTheDocument();
+    expect(getByText('Rozlicz puszkę')).toBeInTheDocument();
+    expect(getByTestId('collected-pln-test-id')).toBeInTheDocument();
+    expect(getByTestId('collected-total-test-id')).toBeInTheDocument();
+  });
+
+  it.concurrent('Testing proper rendering when wosp01 is logged in', ({ expect }) => {
+    const { getByTestId, queryByText, getByText } = renderWithWrapper(
+      <HomePage />,
+      AllRootProvidersWrapper({
+        authContextValues: () => ({
+          ...baseAuthContextValues,
+          username: 'wosp01',
+        }),
+      }),
+    );
+
+    expect(queryByText('Wydaj puszkę wolontariuszowi')).toBeNull();
+    expect(getByText('Rozlicz puszkę')).toBeInTheDocument();
+
+    expect(getByTestId(COLLECTED_PLN_ID)).toHaveTextContent('0 zł');
+    expect(getByTestId(COLLECTED_TOTAL_ID)).toHaveTextContent('0 zł');
+  });
+
+  it.concurrent("Test clicking 'Wydaj puszkę wolontariuszowi' link", ({ expect }) => {
+    const { getByText } = renderWithWrapper(<HomePage />, AllRootProvidersWrapper());
+    const button = getByText('Wydaj puszkę wolontariuszowi');
+    fireEvent.click(button);
```

Comment:
I highly recommend using userEvent, however in this case it does not make any difference